test/e2e/e2e_test.go

```diff
@@ -67,8 +78,29 @@ var _ = BeforeSuite(func() {
 
 	testNamespace = framework.UniqueTestNamespace()
 	createTestNamespace(context.Background())
+
+	initAzureClients()
 })
 
+func initAzureClients() {
+	subscriptionID := os.Getenv(azureSubscriptionEnv)
+	Expect(subscriptionID).ShouldNot(BeEmpty(), "Azure subscription ID is not set")
+
+	atmResourceGroup := os.Getenv(azureTrafficManagerResourceGroupEnv)
+	Expect(atmResourceGroup).ShouldNot(BeEmpty(), "Azure traffic manager resource group is not set")
+
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
```

Comment:
Is this the way to use the federated identity? Do we use the same identity that provisions the aks clusters for existing e2e tests?

Reply:
the login is part of the github action, https://github.com/Azure/fleet-networking/blob/main/.github/workflows/e2e-tests.yml#L66

Yes, we use the same identity for the setup scripts.

This func will automatically use the identity which you've already logged in.